src/ProjectSystem/Impl/IO/MsBuildFileSystemWatcher.cs

```diff
@@ -215,6 +215,9 @@ public sealed partial class MsBuildFileSystemWatcher : IDisposable {
 
             relativePath = PathHelper.MakeRelative(rootDirectory, fullPath);
             try {
+                if (!fileSystem.FileExists(fullPath)) {
```

Comment:
OK, I can remove this if the exception is not causing trouble. CPS exception was caused by Link metadata and appears to go away when I remove Link property from rules. But this is more or less 'best guess'